### PR TITLE
XIVY-14010 implemented green circles

### PR DESCRIPTION
**Pull request by ivy-fhe**
Implements new Design:
image

### CODE REVIEW

**Review by ivy-lli — CHANGES_REQUESTED** [review]
Yes very nice!
One thing, I think now it is not needed that you rebind the default edge to your impl right? Of course currently you add there the segment stuff to the args etc. but I think you should do this all in the mining-action.ts where you add the new 'edge:label:minig'. So I mean set the segments to the args of this new EdgeLabel. Or even create a new MiningEdgeLabel model which extends the EdgeLabel but with the additional attributes, like the position of the label etc.

**Comment by ivy-fhe** [comment]
I don't think that I can access the segments outside the EdgeView, because they are generated in the "render" function of the RoutableView. Maybe I missed something :sweat_smile:

**Comment by ivy-lli** [comment]
Did you check from where this renderAddionals come from, or better from where the segments are coming from?

I'm not sure, as I've not tested it, but I think you can do the same:

```tsx
@inject(EdgeRouterRegistry) edgeRouterRegistry: EdgeRouterRegistry;

... 

const segments = this.edgeRouterRegistry.route(edge, args);
```

File: src/process-mining-visualisation/MiningLabelView.tsx [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+    } else if (value <= 0.75) {
+      props.r = 12;
+      props.color = '#4ccd73';
+    } else if (value <= 1) {
+      props.r = 13;
+      props.color = '#205630';
+      props.textColor = '#ffffff';
+    }
+    return props;
+  };
```
Comment (ivy-lli) [inline src/process-mining-visualisation/MiningLabelView.tsx:55]: please check if this also work in dark mode (add get query param `theme=dark` to URL)

File: src/process-mining-visualisation/mining-action.ts [hunk 2]
```diff
@@ -52,16 +53,17 @@ export class MiningCommand extends Command {
...
-        if (element.args === undefined) {
-          element.args = {};
-        }
-        element.args['labelvalue'] = node.labelvalue;
-        element.args['relativevalue'] = node.relativevalue;
+      const edge = model.index.getById(node.ID);
+      if (edge instanceof Edge) {
+        const segments = this.edgeRouterRegistry.route(edge, edge.args);
+        const miningLabel = new MiningLabel(node.labelvalue, node.relativevalue, segments);
+        this.moveExistingLabel(edge.editableLabel as GLabel, segments);
```
Comment (ivy-lli) [inline src/process-mining-visualisation/mining-action.ts:65]: why do you need to move the existing label?

**Review by ivy-lli — COMMENTED** [review]
Yea 👍

File: src/process-mining-visualisation/MiningLabelView.tsx (hunk 1 above)
Comment (ivy-lli) [reply src/process-mining-visualisation/MiningLabelView.tsx:55]: The point isn't exaclty if it work, but if it looks good ;)
But I would agree 👍

File: src/process-mining-visualisation/mining-action.ts (hunk 2 above)
Comment (ivy-fhe) [reply src/process-mining-visualisation/mining-action.ts:65]: I move the label to try and avoid overlapping:
image
vs
image

but I changed it to only move if the distance between label and mining-annotation to small

**Review by ivy-lli — APPROVED** [review]
Very good work, thanks 👍

**Pull request merged**